### DISCUSSION TITLE
Showbinds show command name

### DISCUSSION THREAD

**Discussion by og900aero** [gokcehan/lf#1853] [Ideas]
It would be nice if showbinds would not only show the command to run, but also an assigned name, because often the command itself is very long and it is opaque what it is, like a name is assigned.

**Comment by DusanLesan** [comment]
I think that would be a good thing to have. Idea was covered in this issue: https://github.com/gokcehan/lf/issues/536

**Reply by og900aero** [reply]
Uhh, it is very old post. Then I wonder why it hasn't been programmed. I've got things so complex programmed that I can't figure out what it does in the list :D

**Reply by DusanLesan** [reply]
Actually, I have just noticed that comments are displayed in the command preview. That is good enough for me if I ever decide to add descriptions.
![image](https://github.com/user-attachments/assets/33f5b175-1e95-4d6f-8c60-86c88275c3d4)

**Reply by og900aero** [reply]
Yes, its working, if the command doesnt too long. But if too long, then you can see a lot of characters, and so it can't be taken out properly if I write a comment too.

**Reply by DusanLesan** [reply]
Do you have any example of such commands? I am just curious why would you make such long commands in the mapping statements. If I write any line of the code that would not fit in one half of the screen (around 130 chars), I break it town or move to a new function

**Reply by og900aero** [reply]
Example:

`map dD $for file in $fx; do if [ -w "$file" ]; then mv "$file" /home/Data/.Trash/; else sudo -A mv "$file" /home/Data/.Trash && sudo -A chown -R $USER:$USER /home/Data/.Trash; fi; done`

**Reply by DusanLesan** [reply]
Oh ok. So you are doing some sort of code minification. Most would write it as a separate `cmd` statement but I guess that in your case that is not ok. Best I can think of is this
![image](https://github.com/user-attachments/assets/d7a2d8d0-1608-44b1-a260-ddf66331f18a)

**Reply by og900aero** [reply]
Yes, that's a good idea. It would look for a description tag after each map command. If it finds it, showbinds will display it, if it doesn't find it, show the command itself.
